.github/workflows/build-libpico.yml

```diff
@@ -0,0 +1,42 @@
+# Run whenever it is manually triggered, a pull request or a push is done that modifes the libpico configuration 
+
+name: libpico Builder
+
+on:
+  pull_request:
+    paths:
+        - tools/libpico/**  
+  workflow_dispatch:
+  push:
+    paths:
+      - tools/libpico/**
+jobs:
+    build-libpico:
+        name: Build libpico precompiled libraries
+        runs-on: ubuntu-latest
+        steps:
+        - uses: actions/checkout@v4
+          with:
+            submodules: false
+        - name: Get submodules for pico-sdk
+          run: cd pico-sdk && git submodule update --init --recursive
+        - name: Install dependencies
+          run: |
+            sudo apt update
+            sudo apt install cmake make build-essential wget gcc-arm-none-eabi
```

Comment:
Oops, I think I confirmed this ain't gonna work since GCC12.4's SDK can't be used with this core.  I've got a patch in newlib and in the core to align the locks for `FILE`s.

There's no way the generic version has allocated the proper space (8 bytes, not 4) or is using the proper lock calls (nor did the GCC12).

You could `wget https://patch-diff.githubusercontent.com/raw/raspberrypi/pico-sdk/pull/2000.patch` and apply it, and then just use `get.py` to install the normal toolchain and no hard-coding needed.  Once we move to 2.0.1 SDK, this patch can be dropped...

Reply:
Oh yeah, that's a good idea! With `get.py`, the toolchain should always be right. I can store the `.patch` file locally and `git apply` it before compilation.

Reply:
Works great! Latest commit now builds with the toolchains from `get.py` and downloads and applies the `2000.patch` file dynamically.

Sadly the commits must be squashed, the merges messed up the history..